contracts/0.8.25/vaults/OperatorGrid.sol

```diff
@@ -0,0 +1,411 @@
+
+        group_.shareLimit = uint96(_shareLimit);
+
+        emit GroupShareLimitUpdated(_nodeOperator, uint96(_shareLimit));
+    }
+
+    /// @notice Returns a group by ID
+    /// @param _nodeOperator address of the node operator
+    /// @return Group
+    function group(address _nodeOperator) external view returns (Group memory) {
+        return _getStorage().groups[_nodeOperator];
+    }
+
+    /// @notice Returns a node operator address by index
+    /// @param _index index of the node operator
+    /// @return Node operator address
+    function nodeOperatorAddress(uint256 _index) external view returns (address) {
+        return _getStorage().nodeOperators[_index];
+    }
+
+    /// @notice Returns a node operator count
+    /// @return Node operator count
+    function nodeOperatorCount() external view returns (uint256) {
+        return _getStorage().nodeOperators.length;
+    }
+
+    /// @notice Registers a new tier
+    /// @param _nodeOperator address of the node operator
+    /// @param _tiers array of tiers to register
+    function registerTiers(
+        address _nodeOperator,
+        TierParams[] calldata _tiers
+    ) external onlyRole(REGISTRY_ROLE) {
+        if (_nodeOperator == address(0)) revert ZeroArgument("_nodeOperator");
+
+        ERC7201Storage storage $ = _getStorage();
+        Group storage group_ = $.groups[_nodeOperator];
+        if (group_.operator == address(0)) revert GroupNotExists();
+
+        uint256 tierIndex = $.tiers.length;
+        uint256 length = _tiers.length;
+        for (uint256 i = 0; i < length; i++) {
+
+            Tier memory tier = Tier({
+                operator: _nodeOperator,
+                shareLimit: uint96(_tiers[i].shareLimit),
+                reserveRatioBP: uint16(_tiers[i].reserveRatioBP),
+                rebalanceThresholdBP: uint16(_tiers[i].rebalanceThresholdBP),
+                treasuryFeeBP: uint16(_tiers[i].treasuryFeeBP),
+                mintedShares: 0
+            });
+            $.tiers.push(tier);
+            group_.tiersId.push(tierIndex);
+
+            emit TierAdded(
+                _nodeOperator,
+                tierIndex,
+                uint96(_tiers[i].shareLimit),
+                uint16(_tiers[i].reserveRatioBP),
+                uint16(_tiers[i].rebalanceThresholdBP),
+                uint16(_tiers[i].treasuryFeeBP)
+            );
+
+            tierIndex++;
+        }
+    }
+
+    /// @notice Registers a new vault
+    /// @param vault address of the vault
+    function registerVault(address vault) external {
+        if (vault == address(0)) revert ZeroArgument("_vault");
+
+        ERC7201Storage storage $ = _getStorage();
+        VaultTier storage vaultTier = $.vaultTier[vault];
+        if (vaultTier.tierIndex > 0) revert VaultExists();
+
+        Group storage defaultGroup = $.groups[DEFAULT_GROUP_ADDRESS];
+        if (defaultGroup.tiersId.length == 0) revert TiersNotAvailable();
```

Comment:
probably can remove if:
- add default tier as a part of the initializer fn
- tiers are editable